api/service/api.go

```
@@ -0,0 +1,240 @@
+		}
+
+		return common.Hash(result.Data.Root), nil
+	} else {
+		a.metrics.RecordBlockIdType(m.BlockIdTypeInvalid)
+		return common.Hash{}, newBlockIdError(id)
+	}
+}
+
+// blobSidecarHandler implements the /eth/v1/beacon/blob_sidecars/{id} endpoint, using the underlying DataStoreReader
+// to fetch blobs instead of the beacon node. This allows clients to fetch expired blobs.
+func (a *API) blobSidecarHandler(w http.ResponseWriter, r *http.Request) {
+	param := chi.URLParam(r, "id")
+	beaconBlockHash, err := a.toBeaconBlockHash(param)
+	if err != nil {
+		err.write(w)
+		return
+	}
+
+	result, storageErr := a.dataStoreClient.Read(r.Context(), beaconBlockHash)
+	if storageErr != nil {
+		if errors.Is(storageErr, storage.ErrNotFound) {
+			errUnknownBlock.write(w)
+		} else {
+			a.logger.Info("unexpected error fetching blobs", "err", storageErr, "beaconBlockHash", beaconBlockHash.String(), "param", param)
+			errServerError.write(w)
+		}
+		return
+	}
+
+	blobSidecars := result.BlobSidecars
+
+	filteredBlobSidecars, err := filterBlobs(blobSidecars.Data, r.URL.Query().Get("indices"))
+	if err != nil {
+		err.write(w)
+		return
+	}
+
+	blobSidecars.Data = filteredBlobSidecars
+	responseType := r.Header.Get("Accept")
+
+	if responseType == sszAcceptType {
+		res, err := blobSidecars.MarshalSSZ()
+		if err != nil {
+			a.logger.Error("unable to marshal blob sidecars to SSZ", "err", err)
+			errServerError.write(w)
+			return
+		}
+
+		_, err = w.Write(res)
+
+		if err != nil {
+			a.logger.Error("unable to write ssz response", "err", err)
+			errServerError.write(w)
+			return
+		}
+	} else {
+		err := json.NewEncoder(w).Encode(blobSidecars)
+		if err != nil {
+			a.logger.Error("unable to encode blob sidecars to JSON", "err", err)
+			errServerError.write(w)
+			return
+		}
+	}
+}
+
+// filterBlobs filters the blobs based on the indices query provided.
+// If no indices or invalid indices are provided, the original blobs are returned.
```

I think it would be better to error out on invalid indices.

---

The code does error -- this was an outdated godoc. Updated 👍

---

(unless the spec says not to....)